pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -92,6 +94,34 @@ public static PinotQuery compileToPinotQuery(String sql)
     return pinotQuery;
   }
 
+  static void validate(PinotQuery pinotQuery)
+      throws SqlCompilationException {
+    // Sanity check group by query: All identifiers in selection list should be also included in group by list.
```

Comment:
Should we check identifiers or expressions?

Reply:
I don't think we should support function expressions part of GROUP BY clause. See example here https://github.com/apache/incubator-pinot/pull/5014#discussion_r370769811

Reply:
GroupBy clause could be non-aggregate expressions, like dateConverter(), ADD(...), SUB(...)